bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/StackFrame.java

```diff
@@ -102,4 +113,36 @@ public StackFrame(PackageInfo packageInfo, CallableUnitInfo callableUnitInfo, Co
         NON_PARTICIPANT
     }
 
+    public void handleChannelPanic(BError error) {
+        for (int i = 0; i < workerSendInChannels.length; i++) {
+            WorkerDataChannel channel;
+            if (workerSendInChannels[i].channelInSameStrand) {
```

Comment:
can we get the value first and do these checks, I think this was done by me, sorry about that, but can we change that?

Reply:
+1